lib/queuePopulator/QueuePopulator.js

```diff
@@ -264,6 +264,48 @@ class QueuePopulator {
         }
         return this._processAllLogEntries(params, done);
     }
+
+    /**
+     * Returns current zookeeper client status.
+     * Possible values are under zookeeper.State.
+     * @returns {string}
+     */
+    zkStatus() {
+        return this.zkClient.getState();
+    }
+
+    /**
+     * Handle ProbeServer liveness check
+     *
+     * @param {import ('http').HTTPServerResponse} res
+     * @param {Logger} log
+     */
+    handleLiveness(res, log) {
+        // log verbose status for all checks
+        const verboseLiveness = {}
+        // track and return all errors in one response
+        let responses = [];
+        const zkState = this.zkStatus();
+        verboseLiveness['zookeeper'] = zkState.code;
+        if (zkState.code !== zookeeper.State.SYNC_CONNECTED.code) {
+            responses.push(`Zookeeper is not connected: ${state}`);
+        }
+        for(const reader of this.logReaders) {
+            const prodStatuses = reader.getProducerStatus();
+            for (const [topic, status] in Object.entries(prodStatuses)) {
+                verboseLiveness[JSON.stringify(reader.getLogInfo())] = status;
+                if (!status) {
+                    responses.push(`log reader topic '${topic}' is not ready`);
+                }
+            }
+        }
+        log.debug('verbose liveness', verboseLiveness)
+        if (responses != []) {
+            return responses.join('. ');
```

Comment:
The idea here is to output any and all errors in one go, instead of piecemeal if many occur at the same time. Similar to how we changed our operator.